azurerm/resource_arm_kubernetes_cluster_test.go

```diff
@@ -742,6 +768,78 @@ resource "azurerm_kubernetes_cluster" "test" {
 `, rInt, location, rInt, rInt, rInt, rInt, rInt, clientId, clientSecret)
 }
 
+func testAccAzureRMKubernetesCluster_addonProfileAciConnectorLinux(rInt int, clientId string, clientSecret string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestvirtnet%d"
+  address_space       = ["172.0.0.0/16"]
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  tags {
+    environment = "Testing"
+  }
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "acctestsubnet%d"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  virtual_network_name = "${azurerm_virtual_network.test.name}"
+  address_prefix       = "172.0.2.0/24"
+}
+
+resource "azurerm_subnet" "test-aci" {
+  name                 = "acctestsubnet-aci%d"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  virtual_network_name = "${azurerm_virtual_network.test.name}"
+  address_prefix       = "172.0.3.0/24"
+}
```

Comment:
based on the failing test it appears that Subnet Delegation needs to be configured here:

```
delegation.#:                                "1" => "0"
delegation.0.name:                           "aciDelegation" => ""
delegation.0.service_delegation.#:           "1" => "0"
delegation.0.service_delegation.0.actions.#: "1" => "0"
delegation.0.service_delegation.0.actions.0: "Microsoft.Network/virtualNetworks/subnets/action" => ""
delegation.0.service_delegation.0.name:      "Microsoft.ContainerInstance/containerGroups" => ""
```

it should be possible to populate these details using:

```
resource "azurerm_subnet" "test-aci" {
  # ...
  delegation {
    name = "aciDelegation"
    service_delegation {
      name = "Microsoft.ContainerInstance/containerGroups"
      actions = [ "Microsoft.Network/virtualNetworks/subnets/action" ]
    }
  }
}
```

at which point this test should pass (I'd question why AKS is modifying the Subnet rather than rejecting it as invalid but ¯\_(ツ)_/¯)

Reply:
Thanks! I missed that in the docs